### ISSUE TITLE
False positive in ExplicitCollectionElementAccessMethodSpec

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4252]
## Expected Behavior
```kotlin
class A {
  fun operator get(key: String): String {
    return key
  }
}

// on tests
val a = mock {
  on { get(any()) } doReturn "hello"
}
```

`get(any())` can't be replaced with the operator `[]`. To do so I would need to write `this[any()]` and I think that's worst than `get(any())`.

That's my use case, but this issue is easier to reproduce than that. For example this code:

```kotlin
fun f() {
   val list = listOf<String>() 
   val value = with(list) { get(0) } 
}
```

It has little sense to write code like that but it is valid kotlin and it shouldn't be flagged.

## Your Environment
* Version of detekt used: 1.19.0-RC1 (this is a regression, it was working ON on 1.18.1)

**Comment by cortinico** [comment]
I believe this was introduced by https://github.com/detekt/detekt/pull/4201
@Whathecode would you be able to jump on this?
